### PR TITLE
ONS-REUK-Service

### PR DESCRIPTION
**Pull request by osamede20**
Rework done on script
Cube class added
Trace annotation added

### CODE REVIEW

File: datasets/ONS-REUK-Service/main.py [hunk 1]
```diff
@@ -12,65 +39,126 @@ def right(s, amount):
...
-tidied_sheets.append(table)
+for tab in tabs:
+    if 'NUTS1 by category' in tab.name:
+        print (tab.name)
+        
+        trace.start(datacube_name, tab, columns, scraper.distributions[0].downloadURL)
+        tab_df = distribution.as_pandas(sheet_name = tab.name)
+        
+        trace.obs("Values from the dataframe")
+        observations = tab_df.iloc[2:17]
```
Comment (mikeAdamss) [inline datasets/ONS-REUK-Service/main.py:96]: I know this wasn't you (and don't worry about changing it in this instance) but try and avoid these kinds of explicit index references in your own work, they tend to be very fragile.

**Review by mikeAdamss — COMMENTED** [review]
On the whole this looks good to me, there's some changes needed around using the scraper and seed but it's an easy tweak and a good thing to know. Happy to approve once those tweaks are in.

Nice one :+1:

File: datasets/ONS-REUK-Service/main.py (hunk 1 above)
Comment (osamede20) [reply datasets/ONS-REUK-Service/main.py:96]: All changes are well acknowledged and noted, and will be effected shortly. Thanks so much.

**Comment by osamede20** [comment]
New commit 748adbf following on from Pull Request review
Legacy code replaced
TransformTrace function applied more efficiently

**Pull request merged**